pkg/api/leader_lease_types_test.go

```diff
@@ -0,0 +1,71 @@
+package api
+
+import (
+	"testing"
+	"time"
+
+	. "github.com/onsi/gomega"
+)
+
+func Test_LeaderLeaseTypes_BeforeCreate(t *testing.T) {
+	tests := []struct {
+		name        string
+		leaderLease *LeaderLease
+	}{
+		{
+			name:        "assigns the id to a leader lease before creation",
+			leaderLease: &LeaderLease{},
+		},
+	}
+
+	RegisterTestingT(t)
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := tt.leaderLease.BeforeCreate(nil)
+			Expect(tt.leaderLease.ID).ToNot(BeEmpty()) // a new id is generated each time, we only care for the ID value to contain something and not the actual value that it holds
+			Expect(err != nil).To(BeFalse())
```

Comment:
Error check should be the first thing. Move this before the other `Expect`.
Moreover, please use the `HaveOccurred()` when checking errors
```suggestion
			Expect(err).ToNot(HaveOccurred())
```